pkg/gofr/container/sql_mock.go

```diff
@@ -41,35 +43,61 @@ type sqlMockDB struct {
 	logger logging.Logger
 }
 
-func (m sqlMockDB) Select(_ context.Context, _ interface{}, query string, args ...interface{}) {
+func (m sqlMockDB) Select(_ context.Context, value any, query string, args ...interface{}) {
 	if len(m.queryWithArgs) == 0 {
-		m.logger.Fatalf("Did not expect any calls for Select with query: %q", query)
+		m.logger.Errorf("did not expect any calls for Select with query: %q", query)
+		return
 	}
 
 	lastIndex := len(m.queryWithArgs) - 1
+
+	defer func() {
+		m.queryWithArgs = m.queryWithArgs[:lastIndex]
+	}()
+
 	expectedText := m.queryWithArgs[lastIndex].queryText
 	expectedArgs := m.queryWithArgs[lastIndex].arguments
 
+	valueType := reflect.TypeOf(value)
+	if valueType.Kind() != reflect.Ptr {
+		m.logger.Errorf("expected a pointer type: %q", query)
+		return
+	}
+
+	if m.queryWithArgs[lastIndex].value == nil {
+		m.logger.Errorf("received different expectations: %v", query)
```

Comment:
```suggestion
		m.logger.Errorf("received different expectations: %q", query)
```

You are using quoted format everywhere else